### PR TITLE
Fix some rescue calls that do not specifiy error type.

### PR DESCRIPTION
**Pull request by utilum**
### What was the end-user problem that led to this PR?

The problem I noticed was in style, several instances of `rescue` clauses that do not specify an exception type. This is noted in in the Ruby Style Guide as [Avoid rescuing the Exception class](https://github.com/bbatsov/ruby-style-guide#no-blind-rescues).

### What was your diagnosis of the problem?

My diagnosis was that at least some of those are leftover style. They are noted in `.rubocop_todo.yml`. To make sure, I asked on Slack.

### What is your fix for the problem, implemented in this PR?

My fix is to make as many of them more specific, specifying at least `StandardError`, and trying to be more specific.

### Why did you choose this fix out of the possible options?

No other ways of addressing this came to mind. I'd be happy to consider.

### CODE REVIEW

**Comment by colby-swandale** [comment]
Hi, please fill out the [PR template](https://github.com/bundler/bundler/blob/master/.github/PULL_REQUEST_TEMPLATE.md).

File: lib/bundler/cli.rb [hunk 1]
```diff
@@ -745,7 +745,7 @@ def warn_on_outdated_bundler
       end
 
       Bundler.ui.warn "The latest bundler is #{latest}, but you are currently running #{current}.\n#{suggestion}"
-    rescue
+    rescue RuntimeArgument
```
Comment (segiddins) [inline lib/bundler/cli.rb]: What is “RuntimeArgument”?

Comment (utilum) [reply lib/bundler/cli.rb]: Naught but a tyop. Fixed.

**Comment by utilum** [comment]
> How did you decide between Runtime and Standard error?
I preferred `RuntimeError` as it is more specific than `StandardError, won't rescue a `SystemCallError` for example.
When using `RuntimeError` caused a failure of the tests and I was not certain which narrower `Exception` may need rescuing, I resorted to `StandardError`. Perhaps those cases are redundant? Would you have them removed?

**Comment by indirect** [comment]
This looks good to me once the tests are green. 👍

**Comment by colby-swandale** [comment]
ping @utilum

**Comment by utilum** [comment]
Sorry for the delay.
I don't have Ruby 1.8.7 locally, and that's where the test failures occur. Just looking at the test log I wonder if the fix to this would not actually be in updating the tests... Anyone?

**Comment by utilum** [comment]
Anyone?

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**